### ISSUE TITLE
Observable props for @storybook/angular are not propagated correctly when knobs used

### ISSUE THREAD

**Issue by victorward** [storybookjs/storybook#10635]
**Describe the bug**
Observable props for `@storybook/angular` are not propagated correctly when knobs are used. If you use knobs together with some _Observable_ parameter then, during runtime, after changing knob value - observable value is reset.

**To Reproduce**
This bug easy to reproduce.

1. Create story
```javascript
[...]
.add('default', () => {
   const boolKnob = boolean('show', true);
   const subj$ = new BehaviorSubject(false);
   const testObs$ = subj$.pipe(
      delay(2000),
      map(() => true)
    );
 return {
      template: `<div>{{ testObs$ | async }}<br/>Knob - {{ boolKnob }}</div>`,
     props: { boolKnob, testObs$ }
}
```
2. After story initialization you will see correct value (`false `or `true`)
3. Change knob checkbox _show_
4. Now `testObs$` is **`null`**

**Expected behavior**
I except that `testObs$` will contain Observable and will be not changed

**System:**

> System:
>     OS: Windows 10 10.0.18363
>   Binaries:
>     Node: 12.14.1 
>     npm: 6.13.4
>   Browsers:
>     Google Chrome - 81.0.4044.129 
>   npmPackages:
>     @storybook/addon-a11y: ^5.3.18 => 5.3.18
>     @storybook/addon-actions: ^5.3.18 => 5.3.18
>     @storybook/addon-centered: ^5.3.18 => 5.3.18
>     @storybook/addon-knobs: ^5.3.18 => 5.3.18
>     @storybook/addon-links: ^5.3.18 => 5.3.18
>     @storybook/addon-notes: ^5.3.18 => 5.3.18
>     @storybook/addon-options: ^5.3.18 => 5.3.18
>     @storybook/addons: ^5.3.18 => 5.3.18
>     @storybook/angular: ^5.3.18 => 5.3.18

**Comment by shilman** [comment]
Hi gang, We’ve just released addon-controls in 6.0-beta!

Controls are portable, auto-generated knobs that are intended to replace addon-knobs long term.

- [Controls README](https://github.com/storybookjs/storybook/tree/next/addons/controls)
- [Migration instructions](https://github.com/storybookjs/storybook/blob/next/addons/controls/README.md#how-do-i-migrate-from-addon-knobs)

Please upgrade and try them out today. Thanks for your help and support getting this stable for release!

**Comment by shilman** [comment]
For anybody who is interested in Controls but don't know where to start, I've created a quick & dirty step-by-step walkthrough to go from a fresh CRA project to a working demo. Check it out:

=> [Storybook Controls w/ CRA & TypeScript](https://gist.github.com/shilman/69c1dd41a466bae137cc88bd2c6ef487)

There are also some "knobs to controls" migration docs in the Controls README:

=> [How do I migrate from addon-knobs?](https://github.com/storybookjs/storybook/blob/next/addons/controls/README.md#how-do-i-migrate-from-addon-knobs)
